.github/actions/check-spelling/action.yml

```diff
@@ -23,25 +21,27 @@ inputs:
       cspell:dotnet/dotnet.txt
       cspell:scala/scala.txt
       lombiq:expect.txt
-
-permissions:
-  contents: read
-  pull-requests: read
-  actions: read
-
-runs-on: ubuntu-latest
+  config:
+    description: Spelling configuration directory
+    required: false
+    type: string
+    default: .github/actions/spelling
+  spell-check-this:
+    description: Repository with default configuration to use
+    required: false
+    type: string
+    default: ''
 
 runs:
   using: "composite"
   steps:
-  - name: check-spelling
-    id: spelling
-    uses: check-spelling/check-spelling@v0.0.20
+  - name: Check Spelling
+    uses: check-spelling/check-spelling@5ca3511e047d3e39719b1a41b56397ecec0e27d5
```

Comment:
> Always pin the action to a specific full-length commit SHA, not a version tag, since [this ensures that we security use a given version](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions). **For readability, also comment the version there.**